sdk/translation/Azure.AI.Translation.Document/tests/Infrastructure/DocumentTranslationLiveTestBase.cs

```diff
@@ -93,7 +93,38 @@ public async Task<Uri> CreateTargetContainerAsync(List<TestDocument> documents =
             var expiresOn = DateTimeOffset.UtcNow.AddHours(1);
             return containerClient.GenerateSasUri(BlobContainerSasPermissions.List | BlobContainerSasPermissions.Write, expiresOn);
         }
+        public async Task<Tuple<Uri, BlobContainerClient>> CreateTargetContainerWithClientAsync(List<TestDocument> documents = default)
+        {
+            Recording.DisableIdReuse();
+            var containerName = "target" + Recording.GenerateId();
+            var containerClient = GetBlobContainerClient(containerName);
+            await containerClient.CreateAsync(PublicAccessType.BlobContainer).ConfigureAwait(false);
 
+            if (documents != default)
+            {
+                await UploadDocumentsAsync(containerClient, documents);
+            }
+
+            var expiresOn = DateTimeOffset.UtcNow.AddHours(1);
+            return Tuple.Create(containerClient.GenerateSasUri(BlobContainerSasPermissions.List | BlobContainerSasPermissions.Write, expiresOn),containerClient);
+        }
+
+        public async Task<Uri> CreateGlossaryAsync(TestDocument document = default)
+        {
+            Recording.DisableIdReuse();
+            var glossaryContainerName = "glossary" + Recording.GenerateId();
+            var containerClient = GetBlobContainerClient(glossaryContainerName);
+            await containerClient.CreateAsync(PublicAccessType.BlobContainer).ConfigureAwait(false);
+
+            if (document != default)
+            {
+                await UploadDocumentsAsync(containerClient, new List<TestDocument> { document });
+            }
+
+            var expiresOn = DateTimeOffset.UtcNow.AddHours(1);
+            var glossaryContainerSasUri = containerClient.GenerateSasUri(BlobContainerSasPermissions.Read | BlobContainerSasPermissions.Write, expiresOn);
```

Comment:
curious, looking at other functions like for example `CreateTargetContainerAsync` we just do `containerClient.GenerateSasUri()...` and return that value. why the need to create a new URI here?

Reply:
Oh, because the glossaryContainerSasUri is used in the following line to build the glossarySasUri, which ends up being the return value of the function.

Reply:
what id we just do:
`return containerClient.GenerateSasUri(BlobContainerSasPermissions.Read | BlobContainerSasPermissions.Write, expiresOn);`

Reply:
Then that would return the SasUri for the container itself, not the SasUri for the "blob", which is the glossary in this case. Other functions like CreateTargetContainer etc. should return the SasUri for the container, since this what is needed as input for creating DocumentTranslationInput. However, to create a TranslationGlossary object, you need the SasUri for the glossary itself.

Reply:
>to create a TranslationGlossary object, you need the SasUri for the glossary itself.

Is that true though? Looking at the documentation, you can pass a container or a file https://docs.microsoft.com/en-us/azure/cognitive-services/translator/document-translation/get-started-with-document-translation?tabs=csharp#create-sas-access-tokens-for-document-translation
so if you just leave it as a container, it should work

Reply:
oh interesting. I hadn't looked at the documentation for this part specifically, I had been following some sample code.

But yea, it does not work. I had tried this before, and I re-tested now to double check and when I do the change that you mentioned, there is no output in the output file. (A blank output file is written). I checked through the Azure portal to make sure that there is in fact a file in the target container, and there was an empty file.

Is this an issue we need to address?